src/ui/server/controllers/insurance/business/turnover/validation/rules/percentage-of-turnover.test.ts

```diff
@@ -0,0 +1,114 @@
+      const expected = generateValidationErrors(PERCENTAGE_TURNOVER, errorMessage, mockErrors);
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe(`when the ${PERCENTAGE_TURNOVER} input contains a letter`, () => {
+    it('should return a validation error', () => {
+      mockBody[PERCENTAGE_TURNOVER] = 'I5';
+      const response = percentageTurnover(mockBody, mockErrors);
+
+      const errorMessage = EXPORTER_BUSINESS[PERCENTAGE_TURNOVER].INCORRECT_FORMAT;
+      const expected = generateValidationErrors(PERCENTAGE_TURNOVER, errorMessage, mockErrors);
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe(`when the ${PERCENTAGE_TURNOVER} input contains a decimal place`, () => {
+    it('should return a validation error', () => {
+      mockBody[PERCENTAGE_TURNOVER] = '3.1';
+      const response = percentageTurnover(mockBody, mockErrors);
+
+      const errorMessage = EXPORTER_BUSINESS[PERCENTAGE_TURNOVER].INCORRECT_FORMAT;
+      const expected = generateValidationErrors(PERCENTAGE_TURNOVER, errorMessage, mockErrors);
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe(`when the ${PERCENTAGE_TURNOVER} input contains a special character`, () => {
+    it('should return a validation error', () => {
+      mockBody[PERCENTAGE_TURNOVER] = '1*';
+      const response = percentageTurnover(mockBody, mockErrors);
+
+      const errorMessage = EXPORTER_BUSINESS[PERCENTAGE_TURNOVER].INCORRECT_FORMAT;
+      const expected = generateValidationErrors(PERCENTAGE_TURNOVER, errorMessage, mockErrors);
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe(`when the ${PERCENTAGE_TURNOVER} input contains a special character and is below 0`, () => {
+    it('should return a validation error', () => {
+      mockBody[PERCENTAGE_TURNOVER] = '-1*';
+      const response = percentageTurnover(mockBody, mockErrors);
+
+      const errorMessage = EXPORTER_BUSINESS[PERCENTAGE_TURNOVER].INCORRECT_FORMAT;
+      const expected = generateValidationErrors(PERCENTAGE_TURNOVER, errorMessage, mockErrors);
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe(`when the ${PERCENTAGE_TURNOVER} input is below 0`, () => {
+    it('should return a validation error', () => {
+      mockBody[PERCENTAGE_TURNOVER] = '-1';
+      const response = percentageTurnover(mockBody, mockErrors);
+
+      const errorMessage = EXPORTER_BUSINESS[PERCENTAGE_TURNOVER].BELOW_MINIMUM;
+      const expected = generateValidationErrors(PERCENTAGE_TURNOVER, errorMessage, mockErrors);
+
+      expect(response).toEqual(expected);
+    });
+  });
+
+  describe(`when the ${PERCENTAGE_TURNOVER} input is above 100`, () => {
+    it('should return a validation error', () => {
+      mockBody[PERCENTAGE_TURNOVER] = '125';
+      const response = percentageTurnover(mockBody, mockErrors);
+
+      const errorMessage = EXPORTER_BUSINESS[PERCENTAGE_TURNOVER].ABOVE_MAXIMUM;
```

Comment:
Can we get the error message like in the actual validation, shorter lines then, so it can be

```js
const errorMessage = ERROR_MESSAGE.ABOVE_MAXIMUM
```